src/main/java/avd/inf/jdm/rentmycar/controller/AuthController.java

```diff
@@ -41,9 +42,10 @@ public AuthController(UserService userService) {
     public ResponseEntity<Object> login(
             @RequestBody(required = true) AuthDTO authDTO
     ){
-        User user = userService.getUserByEmail(authDTO.getEmail());
-        if(user == null || !user.getPassword().equals(authDTO.getPassword())) {
+        Optional<User> user = userService.getUserByEmail(authDTO.getEmail());
+        if(user == null || !user.get().getPassword().equals(authDTO.getPassword())) {
```

Comment:
If you return an `Optional<User>` you can't check for `null` anymore because you will get an empty `Optional` back and that is never a `null`. You should check for `user.isEmpty()` instead in this case.

Reply:
fixed